internal/helm/registry/auth.go

```diff
@@ -30,6 +46,13 @@ func LoginOptionFromSecret(registryURL string, secret corev1.Secret) (registry.L
 		if err != nil {
 			return nil, fmt.Errorf("unable to get authentication data from Secret '%s': %w", secret.Name, err)
 		}
+		// Make sure that the obtained auth config is for the requested host.
+		// When the docker config does not contain the credentials for a host,
+		// the credential store returns an empty auth config.
+		// Refer: https://github.com/docker/cli/blob/v20.10.16/cli/config/credentials/file_store.go#L44
+		if authConfig.ServerAddress != parsedURL.Host {
```

Comment:
Does this work if the secret is created like so:

```
 kubectl create secret docker-registry ghcr \
 --docker-server=https://ghcr.io \
--docker-username=mygithubusername \
--docker-password=mygithubreadtoken
```

Does `ServerAddress` removes the protocol?

Reply:
No, it doesn't. So, it results in a mismatch if the dockerconfig contains scheme.
Need to ensure to parse and extract host from server address when there's scheme present.
Added some extra checks and test case to cover this.
Thanks for pointing it out.